elki/src/test/java/elki/math/statistics/dependence/MCDEDependenceMeasureTest.java

```diff
@@ -0,0 +1,138 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package elki.math.statistics.dependence;
+
+import elki.utilities.ELKIBuilder;
+import elki.utilities.datastructures.arraylike.NumberArrayAdapter;
+import elki.utilities.datastructures.arraylike.DoubleArrayAdapter;
+import elki.utilities.random.RandomFactory;
+import elki.math.statistics.tests.mcde.*;
+import java.util.Random;
+
+import static elki.math.statistics.dependence.MCDEDependenceMeasure.Par.TEST_ID;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static elki.math.statistics.dependence.MCDEDependenceMeasure.Par.M_ID;
+
+import org.junit.Test;
+
+/**
+ * @author Alan Mazankiewicz
+ * @author Edouard Fouché
+ */
+
+public class MCDEDependenceMeasureTest {
+
+  @Test public void runTest() {
+
+    // independent
+    double[] indi1 = new double[1000];
+    double[] indi2 = new double[1000];
+
+    RandomFactory rnd = new RandomFactory(0);
+    Random generator = rnd.getRandom();
+
+    for(int i = 0; i < 1000; i++) {
+      indi1[i] = generator.nextDouble();
+      indi2[i] = generator.nextDouble();
+    }
+
+    // linear
+    double[] lin1 = new double[1000];
+    double[] lin2 = new double[1000];
+
+    for(int i = 0; i < 1000; i++) {
+      lin1[i] = i;
+      lin2[i] = i * 2;
+    }
+
+    // linear with small noise
+    double[] small_noise_lin1 = new double[1000];
+    double[] small_noise_lin2 = new double[1000];
+
+    for(int i = 0; i < 1000; i++) {
+      small_noise_lin1[i] = i;
+      small_noise_lin2[i] = (i * 2) + generator.nextGaussian() * 100;
+    }
+
+    double[] more_noise_lin1 = new double[1000];
+    double[] more_noise_lin2 = new double[1000];
+
+    for(int i = 0; i < 1000; i++) {
+      more_noise_lin1[i] = i;
+      more_noise_lin2[i] = (i * 2) + generator.nextGaussian() * 500;
+    }
+
+    DoubleArrayAdapter adapter = DoubleArrayAdapter.STATIC;
+    MWPTest mwpTest = MWPTest.STATIC;
+
+    MCDEDependenceMeasure mwp = new ELKIBuilder<>(MCDEDependenceMeasure.class) //
+        .with(M_ID, 1000)
+        .with(TEST_ID, mwpTest)
+        .build();
+
+    test_result(indi1, indi2, 0.65, 0.35, 0.67, 0.33, adapter, mwp);
+    test_result(lin1, lin2, 1.0, 0.99, 1.0, 0.97, adapter, mwp);
+    test_result(small_noise_lin1, small_noise_lin2, 1.0, 0.96, 1.0, 0.85, adapter, mwp);
+    test_result(more_noise_lin1, more_noise_lin2, 1.0, 0.85, 1.0, 0.83, adapter, mwp);
+
+    test_rankIndex(adapter, mwpTest);
+  }
+
+  public static void test_result(double[] data1, double[] data2, double strict_upper, double strict_lower, double lax_upper, double lax_lower, NumberArrayAdapter adapter, MCDEDependenceMeasure mwp) {
```

Comment:
Java conventions are to use CamelCase for method names. `testResult` is a bit nondescript, maybe `testTolerance`?

Reply:
Also, since data1 and data2 are double[], there is no benefit in making the adapter a parameter. Its always the standard double[] adapter.